modules/auxiliary/scanner/mssql/mssql_login.rb

```diff
@@ -86,12 +106,37 @@ def run_host(ip)
         credential_core = create_credential(credential_data)
         credential_data[:core] = credential_core
         create_credential_login(credential_data)
-
         print_good "#{ip}:#{rport} - Login Successful: #{result.credential}"
+
+        if create_session?
+          begin
+            mssql_client = result.proof
+            session_setup(result, mssql_client)
+          rescue ::StandardError => e
+            elog('Failed: ', error: e)
+            print_error(e)
+            result.proof.conn.close if result.proof&.conn
+          end
+        end
       else
         invalidate_login(credential_data)
         vprint_error "#{ip}:#{rport} - LOGIN FAILED: #{result.credential} (#{result.status}: #{result.proof})"
       end
     end
   end
+
+  def session_setup(result, client)
+    return unless (result && client)
+    rstream = client.sock
+    my_session = Msf::Sessions::MSSQL.new(rstream, { client: client }) # is cwd right?
```

Comment:
Simon will align the other clients not to use `cwd` as a convention, but `database_name` or something else 😄